File: Calculator/calculator.cpp
#include<iostream>
using namespace std;

int main(){
    double number_1, number_2;
    char operation;

    cout<<"Enter First Number: ";
    cin>>number_1;

    cout<<"\nEnter Second Number: ";
    cin>>number_2;

    cout<<"\nEnter Operation (+, -, *, /): ";
    cin>>operation;

    if(operation == '+'){
        cout<<"\nResult: "<<number_1 + number_2;
    }
    else if(operation == '-'){
        cout<<"\nResult: "<<number_1 - number_2;
    }
    else if(operation == '*'){
        cout<<"\nResult: "<<number_1 * number_2;
    }
    else if(operation == '/'){
        if(number_2 != 0){
            cout<<"\nResult: "<<number_1 / number_2;
        }
        else{
            cout<<"\nError: Division by zero is not allowed.";
        }
    }
    else{
        cout<<"\nError: Invalid operation.";
    }
}

File: Calculator/calculator.java
import java.util.Scanner;

class Calculator{
    public static void main(String[] args){
        Scanner number_1 = new Scanner(System.in);
        System.out.print("Enter first number: ");
        double first_num = number_1.nextDouble();

        Scanner number_2 = new Scanner(System.in);
        System.out.print("Enter second number: ");
        double second_num = number_2.nextDouble();
        
        Scanner Operator = new Scanner(System.in);
        System.out.print("Enter Operation (+, -, *, /): ");
        char operator = Operator.next().charAt(0);

        if(operator == '+'){
            System.out.println(first_num + second_num);
        }
        else if(operator == '-'){
            System.out.println(first_num - second_num);
        }
        else if(operator == '*'){
            System.out.println(first_num * second_num);
        }
        else if(operator == '/'){
            if(second_num == 0){
                System.out.println("Division by zero is not allowed.");
            } else {
                System.out.println(first_num / second_num);
            }
        }
        else{
            System.out.println("Invalid Operator");
        }
    }
}